### DISCUSSION TITLE
STATUS_STACK_OVERFLOW when adding more widgets to .slint tree

### DISCUSSION THREAD

**Discussion by maxenko** [slint-ui/slint#5058] [Show and tell]
I just wanted to leave this here if you're running into this issue while adding more widgets to your application UI in slint:
```
Exception 0xc00000fd encountered at address 0x7ff7e999f5e7: Stack overflow
```
it can also look like this:
```
thread 'main' has overflowed its stack
error: process didn't exit successfully: `target\debug\MyAwesome.exe debug` (exit code: 0xc00000fd, STATUS_STACK_OVERFLOW)
```

This error won't show up in slint preview window in vs.code. It will show up while running a release or debug. I've spent hours on it today.

The problem is that (i assume) slint is fairly stack demanding, so the application is blowing past windows default stack limit of 1mb. I can go back to work now that I've increased my stack size via:

```
[target.x86_64-pc-windows-msvc]
rustflags = [
    "-C", "link-arg=/STACK:2000000"
]
```
in 
```
my_project/.cargo/config.toml
```
changing it back to 1000000 makes the problem come back, so its definitely stack size issue. The application is fairly small, maybe less than a hundred widgets all together, kind of surprising.

Hope this saves someone time.

**Comment by ogoffart** [comment]
This reminds me of #133 or more recently #4586

We handle fairly large structs, and it seems like Rust tend to make a few copies on the stack in debug mode which leads to stack overflow.

**Comment by FaintentDev** [comment]
lifesaver 🙏

wondering if this workaround could cause any unintended side effects? not knowing what a stack is

**Reply by maxenko** [reply]
Look up stack vs heap. Its fundamental to memory management in system and many other languages.

**Reply by tronical** [reply]
> lifesaver 🙏
> 
> wondering if this workaround could cause any unintended side effects? not knowing what a stack is

No worries. Your machine has likely plenty of ram and the defaults MSVC applies to the stack size are just too small for use-cases like these.
